spec/features/restaurant_spec.rb

```diff
@@ -66,6 +67,30 @@
     expect(page).to have_content("Delivery")
     expect(page).to have_content(restaurant.delivery)
   end
+
+  scenario "user searches for a restaurant by name" do
+    visit restaurants_path
+    fill_in "search_text", with: review.restaurant.name
```

Comment:
consider a local variable here that makes it easy to identify the restaurant. You could also similarly extract a let like so.

```ruby
let(:restaurant) { review.restaurant }
```